shiny/ui/_card.py

```diff
@@ -143,7 +147,7 @@ def _card_impl(
         },
         *children,
         attrs,
-        _full_screen_toggle() if full_screen else None,
+        _full_screen_toggle(str(attrs["id"])) if full_screen else None,
```

Comment:
This was a type thing; `_full_screen_toggle()` takes an `id: str`, the type checker was complaining that `attrs["id"]` is a `TagAttr` and not a string.

What do you mean by "the attr should be skipped if the value is `None`"?

Reply:
I fixed this by having `_full_screen_toggle()` take a `TagAttrValue`.

Reply:
I fixed this by having `_full_screen_toggle()` take a `TagAttrValue`.

Reply:
If we _know_ it's a `str` and we're just trying to satisfy the type checker, then I think it would be more appropriate to use `cast(str, attrs["id"])`, because the `cast()` function exists just to tell the type checker that an object has a certain type. (Or, in some cases, it can make sense to do a `pyright: ignore`.)

However, if we actually want to do a run-time type-conversion (like if the value could be something other than a string), then it makes sense to use `str()`.

Reply:
Thanks, that makes sense. In this case the value is a `TagAttrValue`, so I think the right fix is to tell `_full_screen_toggle()` that its input will be a `TagAttrValue`. 

I was initially thinking that a `TagAttrValue` is an awkward input type for `_full_screen_toggle()`, which is why I used `str`. But it's an internal helper function, so the awkwardness isn't as bad.